### PR TITLE
Team statistics

### PR DESCRIPTION
**Pull request by cowens87**
PR Notes -->
Completed the following Methods/Tests:

Team_info
Best_season
Worst_season
Average_win_percentage
Most_goals_scored
Fewest_goals_scored
Favorite_opponent
Rival

### CODE REVIEW

File: lib/team_statistics.rb [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    @stat_tracker = stat_tracker
+  end
+# [x] 1. Team Info
+  def team_data_set
+    @stat_tracker[:teams]["team_id"].zip(@stat_tracker[:teams]["franchiseId"], @stat_tracker[:teams]["teamName"], @stat_tracker[:teams]["abbreviation"], @stat_tracker[:teams]["link"])
+  end
+
+  def match_team_id(team_id)
+    team_data_set.map do |info|
+      return info if info[0] == team_id
```
Comment (katemorris) [inline lib/team_statistics.rb:14]: I think you can use select here rather than map, take the `return info if` out.

File: lib/team_statistics.rb [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    end
+  end
+
+  def team_info(team_id)
+    team_data = {}
+    headers = ["team_id", "franchiseId", "teamName", "abbreviation", "link"]
+
+    headers.each_with_index do |header, index|
```
Comment (katemorris) [inline lib/team_statistics.rb:22]: Good find on each with index!

File: lib/team_statistics.rb [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    @stat_tracker[:game_teams]["game_id"].zip(@stat_tracker[:game_teams]["team_id"], @stat_tracker[:game_teams]["result"])
+  end
+
+  def total_games_by_season(team_id)
+    total_season_games = {}
+    # returns for team_id '3' {:total_2012=>2, :total_2016=>0, :total_2014=>1}
+    total_2012 = season_win_data_set.find_all do |game|
+      game[0].start_with?("2012") && game[1] == team_id
+    end.count
+    season_win_data_set.map do |game|
```
Comment (katemorris) [inline lib/team_statistics.rb:39]: We will be able to refactor this down soon, great job figuring this out!

File: lib/team_statistics.rb [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+      end.count
+    season_win_data_set.map do |game|
+      if game[0].start_with?("2016")
+          winning_games_by_season["20162017"] = win_results_2016
+      end
+    end
+    winning_games_by_season
+  end
+
+  def total_loss_by_season(team_id)
```
Comment (katemorris) [inline lib/team_statistics.rb:125]: When we get to refactoring, we can combine these three by passing an argument, just adding that note here.

File: lib/team_statistics.rb [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+
+  def winning_games(team_id)
+    #returns for team_id '6' = 4.0
+    season_win_data_set.find_all do |team|
+      team[1] == team_id && team[2] == "WIN"
+    end.count.to_f
+  end
+
+  def average_win_percentage(team_id)
+    "#{(winning_games(team_id)/total_games(team_id) * 100).round(2)}"
```
Comment (katemorris) [inline lib/team_statistics.rb:203]: That's a cool way to return a string!

File: lib/team_statistics.rb [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+    end
+    mama_bear.map do |opp_team_id, wins|
+      mama_bear[opp_team_id] = (wins.to_f/papa_bear[opp_team_id])
+    end
+    teeny_bear = mama_bear.max_by {|opp_team_id, win_percentage| win_percentage}[0]
+    team_id_to_team_name(teeny_bear)
+  end
+
+  def rival(team_id)
+    papa_bear = {}
```
Comment (katemorris) [inline lib/team_statistics.rb:275]: Great work on this, we will need to refactor these together, but that is what is next. Mama bear, Pape bear, and baby bear forever! Oh and teeny bear.

File: test/team_statistics_test.rb [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      teams: team_path,
+      game_teams: game_teams_path
+    }
+
+    @stat_tracker = StatTracker.from_csv(locations)
+    @game_statistics = GameStatistics.new(@stat_tracker)
+    @team_statistics = TeamStatistics.new(@stat_tracker)
+  end
+
+  # Each of the methods below take a team id as an argument. Using that team id,
```
Comment (katemorris) [inline test/team_statistics_test.rb:25]: At the end we'll need to remove the comments and spaces if there are extra. No worries right now though!

**Review by katemorris — CHANGES_REQUESTED** [review]
On line 14, can you try the change I suggested? Once done, let me know if it worked and I'll approve after that :)

File: lib/team_statistics.rb (hunk 1 above)
Comment (cowens87) [reply lib/team_statistics.rb:14]: return with select: [["12", "26", "Sky Blue FC", "SBL", "/api/v1/teams/12"]] but removing the 'return info if' returns this: {"team_id"=>["1", "23", "Atlanta United", "ATL", "/api/v1/teams/1"], "franchiseId"=>nil, "teamName"=>nil, "abbreviation"=>nil, "link"=>nil}
Left original code. We may be able to tweak this to work.

**Review by katemorris — APPROVED** [review]

**Pull request merged**